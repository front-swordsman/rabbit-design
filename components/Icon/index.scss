@import '../styles/themes/default';
$icon: 'snake-icon';

.#{$icon} {
  width: 1em;
  height: 1em;
  vertical-align: -0.15em;
  fill: currentColor;
  overflow: hidden;
  font-size: 24px;

  &-spin {
    animation: loadingSpin 1s infinite linear;
  }
}

@keyframes loadingSpin {
  100% {
    transform: rotate(360deg);
  }
}
