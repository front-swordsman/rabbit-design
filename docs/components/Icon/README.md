---
imports:
  import IconDemoCode from '!raw-loader!./simpleIcon.tsx';
  import IconDemo from './simpleIcon.tsx';
  import Code from '../../layout/prism/index'
---

# Icon

图标类型的组件，现在是 `svg`，但是 `svg` 在 `IE9` 是不兼容的，目前是一次性全部加载，后续会考虑`按需加载`

## Usage 用法

```jsx
<Icon type="eye">
```

`type` 是必填字段

## Api

| 属性 | 说明 | 类型 | 默认值 | 可选 |
|-----|:-:|:-:| ------: | :-: |
| type | 图标的名称 | string | -- | 否
| size | 图标的尺寸 | number | -- | --
| color | 图标的颜色 | string | -- | --
| rotate | 旋转的角度 | number | -- | --
| spin | 是否渲染 | boolean | `false` | --
| className | 类名 | string | -- | --
| style | 样式 | object | -- | --
| onClick | 点击事件 | function(e) : void | -- | --

## Demo 示例

:::demo
<IconDemo />
:::