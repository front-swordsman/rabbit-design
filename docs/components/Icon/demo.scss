.snake-icon-demo {
  color: #666;
  ul {
    padding: 0;
  }
  li {
    display: inline-block;
    width: 150px;
    text-align: center;
    cursor: pointer;
    transition: 0.3s ease-in-out;
    border-radius: 4px;
    &:hover {
      background-color: #ECF5FD;
      transform: scale(1.2)
    }
  }
  &-icon {
    padding: 20px;
  }
  &-name {
    padding: 10px;
  }
}