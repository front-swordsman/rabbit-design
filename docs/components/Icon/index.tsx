import * as React from 'react'
import IconDoc from './README.md'

export default function IconView() {
  return <IconDoc />
}