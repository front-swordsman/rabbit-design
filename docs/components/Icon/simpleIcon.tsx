import * as React from 'react'
import Icon from 'components/Icon'
import { icons } from './iconList'

import './demo.scss'

export default function IconDemo() {
  return (
    <div className="snake-icon-demo">
      <ul>
        {icons.map((n) => {
          return (
            <li key={n}>
              <div className="snake-icon-demo-icon"><Icon type={n} /></div>
              <div className="snake-icon-demo-name">{n}</div>
            </li>
          )
        })}
      </ul>
    </div>
  )
}