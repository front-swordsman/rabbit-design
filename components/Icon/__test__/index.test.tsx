import * as React from 'react'
import { render, fireEvent } from 'react-testing-library'
import Icon from '../index'

describe('icon test', () => {
  it('snapshot', () => {
    const { container } = render(<Icon type="eye-close" />)
    const { container: contaner1 } = render(
      <Icon type="eye-close" size={20} color="red" rotate={90} spin={false} />
    )
    const { container: contaner2 } = render(<Icon type="eye-close" prefixCls="uni-design" />)
    expect(container.innerHTML).toMatchSnapshot()
    expect(contaner1.innerHTML).toMatchSnapshot()
    expect(contaner2.innerHTML).toMatchSnapshot()
  })
  it('fire click', () => {
    const fn = jest.fn()
    const { container } = render(<Icon type="eye-close" onClick={fn} />)
    fireEvent.click(container.firstChild as HTMLElement)
    expect(fn).toBeCalled()
  })
})
