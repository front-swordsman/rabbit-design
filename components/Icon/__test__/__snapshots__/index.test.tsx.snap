// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon test snapshot 1`] = `"<svg class=\\"snake-icon snake-icon-eye-close\\"><use xlink:href=\\"#icon-eye-close\\"></use></svg>"`;

exports[`icon test snapshot 2`] = `"<svg class=\\"snake-icon snake-icon-eye-close\\" style=\\"font-size: 20px; color: red;\\"><use xlink:href=\\"#icon-eye-close\\"></use></svg>"`;

exports[`icon test snapshot 3`] = `"<svg class=\\"uni-design uni-design-eye-close\\"><use xlink:href=\\"#icon-eye-close\\"></use></svg>"`;
